### PR TITLE
Re-enable carousel integration tests

### PR DESCRIPTION
**Pull request by dvoytenko**
## Technical considerations

This pull requests contains a few minor changes (e.g. it creates a test container for the carousel), but also one important consideration to guard against flakes - the `Container.waitReady` API. Some components such as carousel use `ResizeObserver` to complete their layout and testing before it happens is a big source of flakes. In other words - the layout of such components is asynchronous and unobservable to the test. `waitReady` takes care of it, by allowing the component to declare that it needs time to finish its layout and when it actually happens.

For now, a test should "know" and use this API when testing a relevant feature. I think this is ok since most of components have a good auto-layout, and this is more of an exception.

### CODE REVIEW

**Comment by swissspidy** [comment]
Speaking of readiness.. It looks like there can be false positives when videos or images on the library haven't fully loaded yet. Example:

https://percy.io/google/web-stories-wp/builds/5863854

Screenshot 2020-06-26 at 14 20 11
Screenshot 2020-06-26 at 14 20 04

**Comment by dvoytenko** [comment]
> Speaking of readiness.. It looks like there can be false positives when videos or images on the library haven't fully loaded yet. Example:

@swissspidy Would you mind filing a bug for this? This is somewhat of a different type of readiness and happens on a different process. But I'm pretty surprised this happens since Percy was supposed to wait for `window.onload` that should be taken care of this...

**Comment by swissspidy** [comment]
Sure thing, will create one :-)

Edit: #2787

File: assets/src/edit-story/karma/fixture/containers/carousel.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+export class Carousel extends Container {
+  constructor(node, path) {
+    super(node, path);
+  }
+
+  get pages() {
+    const pageList = this.getByRole('listbox', { name: 'Pages List' });
+    if (!pageList) {
+      return null;
+    }
```
Comment (barklund) [inline assets/src/edit-story/karma/fixture/containers/carousel.js]: Would it make sense to move this to helper function as it's used twice?

**Review by barklund — APPROVED** [review]
LGTM

File: assets/src/edit-story/karma/fixture/containers/carousel.js (hunk 1 above)
Comment (dvoytenko) [reply assets/src/edit-story/karma/fixture/containers/carousel.js]: Done. One actually is easily expressed via the other one.

**Pull request merged**